test/e2e/performanceprofile/functests/utils/profilesupdate/profile_update.go

```diff
@@ -56,7 +54,15 @@ func ApplyProfile(profile *performancev2.PerformanceProfile) error {
 	mcps.WaitForCondition(performanceMCP.Name, mcv1.MachineConfigPoolUpdated, corev1.ConditionTrue)
 
 	testlog.Info("Applying changes in performance profile and waiting until mcp will start updating")
-	profiles.UpdateWithRetry(profile)
+	spec, err := json.Marshal(profile.Spec)
+	Expect(err).ToNot(HaveOccurred())
+	Expect(testclient.Client.Patch(context.TODO(), profile,
+		client.RawPatch(
+			types.JSONPatchType,
+			[]byte(fmt.Sprintf(`[{ "op": "replace", "path": "/spec", "value": %s }]`, spec)),
+		),
+	)).ToNot(HaveOccurred())
```

Comment:
I'm not thrilled in adding expectations inside a utility method in this module. Actually I'm not thrilled about a patchwork approach, we should aim for some consistency.
But even before: why is Patch more reliable than Update?

Reply:
I don't have answers now as to why patch works better than updateWithRetry except the issue is reproducible in this test case everytime. Also in the cases where we do use updateWithRetry , we have seen cases where tests do fail with profile updates fail .  I will investigate on this.  I can remove the exceptions and return err.